woocommerce/woocommerce
# Reset Product Collection Pagination When Filtering

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

While it's true that the Product Collection block uses client-side refreshes, it is still vulnerable to the 404 thrown when a page does not exist. When the filters are changed while the block is looking at a page that no longer exists it will do a full page reload to a 404 page. This pull request adds a page reset so that it will return to the first page whenever the filters are changed.

Closes #45644.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Add a Product Collection block and various filters to the page. Make sure you have more than one page of products by default.
2. Navigate to a different page and adjust the filters.
3. When you change the filter it should navigate back to the first page.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
